jetson_build.sh

```diff
@@ -0,0 +1,109 @@
+            libboost-dev \
+            libh2o-dev \
+            libh2o-evloop-dev \
+            libnuma-dev \
+            libwslay-dev \
+            libuv1-dev && \
+    if [ $(cat /etc/os-release | grep 'VERSION_ID="16.04"' | wc -l) -ne 0 ]; then \
+        apt-get install -y --no-install-recommends \
+                libcurl3-dev; \
+    elif [ $(cat /etc/os-release | grep 'VERSION_ID="18.04"' | wc -l) -ne 0 ]; then \
+        apt-get install -y --no-install-recommends \
+                libcurl4-openssl-dev \
+                zlib1g-dev; \
+    else \
+        echo "Ubuntu version must be either 16.04 or 18.04" && \
+        exit 1; \
+    fi && \
+    rm -rf /var/lib/apt/lists/*
+
+# TensorFlow libraries. Install the monolithic libtensorflow_trtis and
+# create a link libtensorflow_framework.so -> libtensorflow_trtis.so so
+# that custom tensorflow operations work correctly. Custom TF
+# operations link against libtensorflow_framework.so so it must be
+# present (and its functionality is provided by libtensorflow_trtis.so).
+# TODO Copy libtensorflow_trtis.so.1 from artifact store
+(cd /opt/tensorrtserver/lib && \
+    ln -sf libtensorflow_trtis.so.1 libtensorflow_framework.so.1 && \
+    ln -sf libtensorflow_trtis.so.1 libtensorflow_framework.so && \
+    ln -sf libtensorflow_trtis.so.1 libtensorflow_trtis.so && \
+    ln -sf libtensorflow_trtis.so.1 libtensorflow_cc.so)
+
+LIBCUDA_FOUND=$(ldconfig -p | grep -v compat | awk '{print $1}' | grep libcuda.so | wc -l) && \
+    if [[ "$LIBCUDA_FOUND" -eq 0 ]]; then \
+        export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64/stubs; \
+        ln -fs /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/stubs/libcuda.so.1; \
+    fi && \
+    echo $LD_LIBRARY_PATH && \
+    cd ${HOME}/trtis/tensorrt-inference-server
+    rm -fr builddir && mkdir -p builddir && \
+    (cd builddir && \
+            cmake -DCMAKE_BUILD_TYPE=Release \
+                  -DTRTIS_ENABLE_METRICS=OFF \
+                  -DTRTIS_ENABLE_TRACING=OFF \
+                  -DTRTIS_ENABLE_GCS=OFF \
+                  -DTRTIS_ENABLE_S3=OFF \
+                  -DTRTIS_ENABLE_CUSTOM=ON \
+                  -DTRTIS_ENABLE_TENSORFLOW=ON \
+                  -DTRTIS_ENABLE_TENSORRT=OFF \
+                  -DTRTIS_ENABLE_CAFFE2=OFF \
+                  -DTRTIS_ENABLE_ONNXRUNTIME=OFF \
+                  -DTRTIS_ENABLE_ONNXRUNTIME_OPENVINO=OFF \
+                  -DTRTIS_ENABLE_PYTORCH=OFF \
+                  -DTRTIS_EXTRA_LIB_PATHS="/opt/tensorrtserver/lib" \
+                  ../build && \
+            make -j16 trtis && \
+            mkdir -p /opt/tensorrtserver/include && \
+            cp -r trtis/install/bin /opt/tensorrtserver/. && \
+            cp -r trtis/install/lib /opt/tensorrtserver/. && \
+            cp -r trtis/install/include /opt/tensorrtserver/include/trtserver)
+
+# TODO publish this to a fixed data repository 
+tar -zcvf tensorrtserver${TRTIS_VERSION}.tgz /opt/tensorrtserver
```

Comment:
Should have "jetpack-x.y.z" or similar in the filename. Can you find the version of jetpack in some envvar on the build system?

Reply:
They do not store the jetpack version in an environment variable
`echo $(dpkg-query --showformat='${Version}' --show nvidia-l4t-core)`
> 32.3.1-20191209230245

From this script [here](https://github.com/dusty-nv/jetson-inference/blob/7e81381a96c1ac5f57f1728afbfdec7f1bfeffc2/tools/install-pytorch.sh#L321) I was able to get the JetPack version

Reply:
Followed the following [script](https://github.com/jetsonhacks/jetsonUtilities/blob/7623cb3b37aca11ff8f5e3bd0f63d4f595b4d21e/scripts/jetson_variables#L91) for now.